src/uct/cuda/cuda_ipc/cuda_ipc_cache.c

```diff
@@ -108,14 +122,64 @@ static void uct_cuda_ipc_cache_invalidate_regions(uct_cuda_ipc_cache_t *cache,
               cache->name, from, to);
 }
 
-ucs_status_t uct_cuda_ipc_unmap_memhandle(void *rem_cache, uintptr_t d_bptr,
+static
+ucs_status_t uct_cuda_ipc_get_rem_cache(pid_t pid, uct_cuda_ipc_cache_t **cache)
+{
+    ucs_status_t status = UCS_OK;
+    char target_name[64];
+    uct_cuda_ipc_cache_hash_key_t key;
+    khiter_t khiter;
+    int khret;
+
+    ucs_recursive_spin_lock(&remote_cache.lock);
+
+    key.pid       = pid;
+    key.tid       = pthread_self();
```

Comment:
1. too many spaces
2. key should be device id, not thread id

Reply:
Device id is part of the key already but I've removed thread id. Two threads of the same process should be able to use the same remote cache.